File: lib/api/sample_clients/perl/clientGetTestCase.pl
#!/usr/bin/perl
#
# TestLink Open Source Project - http://testlink.sourceforge.net/
# This script is distributed under the GNU General Public License 2 or later.
#
# Filename $RCSfile: clientGetTestCase.pl,v $
#
# ------------------------------------------------------------------------------
#
# @version $Revision: 1.1 $
# @modified $Date: 2010/11/24 18:05:29 $ by $Author: franciscom $
# @Author: 
#
# rev: 
#
#
use RPC::XML::Client;

$devkey='DEVKEY';
$client = RPC::XML::Client->new('http://SERVERNAME/testlink/lib/api/xmlrpc.php');
$testplan_id = XY;

$tcinfo=$client->send_request('tl.getTestCase', { devKey=>$devkey, testcaseexternalid=>'XY-1' });
@resp_array = @{$tcinfo->value};
$tc_id = $resp_array[0]->{"testcase_id"};

$data = { "devKey"=>$devkey, "testcaseid"=>$tc_id, "testplanid"=>$testplan_id, "status"=>"p", "notes"=>"perl example 1", "guess"=>"TRUE" };
$tcinfo=$client->send_request('tl.reportTCResult', $data);
@resp_array = @{$tcinfo->value};
$b  = $resp_array[0];
foreach my $k (keys(%$b)) { print "key \'$k\' has value  \'$b->{$k}\'\n" }


$tcinfo=$client->send_request('tl.getTestCase', { devKey=>$devkey, testcaseexternalid=>'XY-2' });
@resp_array = @{$tcinfo->value};
$tc_id = $resp_array[0]->{"testcase_id"};

$data = { "devKey"=>$devkey, "testcaseid"=>$tc_id, "testplanid"=>$testplan_id, "status"=>"f", "notes"=>"perl example 2", "guess"=>"TRUE" };
$tcinfo=$client->send_request('tl.reportTCResult', $data);
@resp_array = @{$tcinfo->value};
$b  = $resp_array[0];
foreach my $k (keys(%$b)) { print "key \'$k\' has value  \'$b->{$k}\'\n" }

File: lib/api/xmlrpc/v1/sample_clients/php/clientGetTestCase.php
<?php
 /**
 * TestLink Open Source Project - http://testlink.sourceforge.net/
 * This script is distributed under the GNU General Public License 2 or later.
 *
 * Filename $RCSfile: clientGetTestCase.php,v $
 *
 * @version $Revision: 1.1 $
 * @modified $Date: 2009/05/21 20:28:39 $ by $Author: franciscom $
 * @Author: francisco.mancardi@gmail.com
 *
 * rev: 
 */
 
require_once 'util.php';
require_once 'sample.inc.php';
show_api_db_sample_msg();

$devKey = md5('admin');
$method='getTestCase';
$test_num=0;

$args=array();
$args["devKey"]=isset($_REQUEST['apiKey']) ? $_REQUEST['apiKey'] : DEV_KEY;

$args["testcaseexternalid"]='AF-2';
$additionalInfo='';

$debug=true;
$test_num++;
$unitTestDescription="Test {$test_num} - {$method}";
echo $unitTestDescription;
echo $additionalInfo;

$client = new IXR_Client($server_url);
$client->debug=$debug;

runTest($client,$method,$args,$test_num);
// ---------------------------------------------------------------------------------


$args=array();
$args["devKey"]=isset($_REQUEST['apiKey']) ? $_REQUEST['apiKey'] : DEV_KEY;
$args["testcaseexternalid"]='API-2';
$args["version"]=1;
$additionalInfo='';

$debug=true;
$test_num++;
$unitTestDescription="Test {$test_num} - {$method}";
echo $unitTestDescription;
echo $additionalInfo;

$client = new IXR_Client($server_url);
$client->debug=$debug;

runTest($client,$method,$args);
// ---------------------------------------------------------------------------------

$args=array();
$args["devKey"]=isset($_REQUEST['apiKey']) ? $_REQUEST['apiKey'] : DEV_KEY;
$args["testcaseid"]='1667';
$args["version"]=1;
$additionalInfo='';

$debug=true;
$test_num++;
$unitTestDescription="Test {$test_num} - {$method}";
echo $unitTestDescription;
echo $additionalInfo;

$client = new IXR_Client($server_url);
$client->debug=$debug;

runTest($client,$method,$args);
// ---------------------------------------------------------------------------------

$args=array();
$args["devKey"]=isset($_REQUEST['apiKey']) ? $_REQUEST['apiKey'] : DEV_KEY;
$args["testcaseexternalid"]='API-2';
$args["version"]=3;
$additionalInfo='';

$debug=true;
$test_num++;
$unitTestDescription="Test {$test_num} - {$method}";
echo $unitTestDescription;
echo $additionalInfo;

$client = new IXR_Client($server_url);
$client->debug=$debug;

runTest($client,$method,$args);
// ---------------------------------------------------------------------------------
